docs/notebooks/flax_basics.ipynb

```diff
@@ -0,0 +1,911 @@
+{
```

Comment:
<p>Could you explain a bit what happens behind the scenes when w call <code>.apply()</code>  and why we can't call <code>module()</code>  directly?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/flax/pull/542/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Reply:
<p>Do you mean why something like:</p><p><code>model = nn.Dense(features=5)</code></p><p><code>model(x) # Error</code></p><p>doesn't work vs using apply?</p>

Reply:
Exactly. I think this is not obvious for first time users.